src/sidebars/favourites/favourite-controller.js

```diff
@@ -87,7 +92,7 @@
 
                 self.update();
 
-                $scope.$on('updateFavourites', function(event, data) {
+                window.addEventListener('updateFavourites', function(event) {
```

Comment:
To minimise window accesses/the diff, could you add the `updateFavourites` event listener once, and on the callback fire a `$rootScope.$broadcast` in the store service?